stdlib/REPL/src/REPL.jl

```diff
@@ -42,6 +42,13 @@ _displaysize(io::IO) = displaysize(io)::Tuple{Int,Int}
 include("Terminals.jl")
 using .Terminals
 
+"""
+    AbstractREPL
+
+The abstract base type for all REPL implementations in Julia. `AbstractREPL` serves as a blueprint for defining various REPL interfaces.
+
+Implementations of `AbstractREPL` must define specific methods that dictate how the REPL reads input, evaluates it, and prints output.
```

Comment:
Similar to the above, I don't think this is particularly useful either. In general abstract types should either document the interface (which is not really stable here, so probably doesn't make sense) or tell you which concrete type you should choose instead, so some of the feature comparison discussion below might help.

There could also potentially be discussions of the overall REPL architecture (frontend/backend split) or other topics that span across implementations here.